frb_example/benches/dart/lib/main.dart

```diff
@@ -0,0 +1,119 @@
+import 'dart:convert';
+
+import 'package:flutter_rust_bridge/flutter_rust_bridge.dart';
+import 'package:flutter_rust_bridge_benchmark/bridge_definitions.dart';
+import './seeder.dart';
+import 'package:logging/logging.dart';
+import 'package:uuid/uuid.dart';
+import 'ffi.io.dart' if (dart.library.html) 'ffi.web.dart';
+
+void main(List<String> args) async {
+  String dylibPath = args[0];
+  // on web, env useJSON is unset, but server will send a --json cli arg instead
+  // this is because WASM cannot be fed any env var, but happily takes cli args.
+  final allowJSON = useJSON || args.contains('--json');
+  final log = Logger('frb_example/benches');
+  if (allowJSON) {
+    Logger.root.level = Level.INFO;
+  } else {
+    Logger.root.level = Level.ALL;
+  }
+  Logger.root.onRecord.listen((record) {
+    print('${record.level.name}: ${record.time}: ${record.message}');
+  });
+  log.fine('flutter_rust_bridge_benchmark_suite example program start (dylibPath=$dylibPath)');
+  log.finer('construct api');
+  if (isSilicon) {
+    log.warning("⚠️  [env] SILICON: uses open method for dynamic library");
+  }
+  if (useJSON) {
+    log.info("🗃️  [env] JSON: switch to json output");
+  }
+  if (isWeb && !allowJSON) {
+    log.warning("⚠️  on Web, Rust logs to stdout are redirected to browser's console (or simply choose JSON)");
+  }
+  final api = initializeBenchExternalLibrary(dylibPath, useJSON: allowJSON);
+
+  // here freely do as much setup as you need,
+  // since only calls to bridge-wired functions will record any metric
+  final List<UuidValue> thousandUuids = seed(UuidSeeder(), just_1000);
+  final List<String> thousandStrings = seed(StringSeeder(), just_1000);
+  final List<UuidValue> hundredThousandUuids = seed(UuidSeeder(), just_100_000);
+  final List<String> hundredThousandStrings = seed(StringSeeder(), just_100_000);
+
+  // real game starts here
+  // --- (benchmarks:start) (avoid any initialization or computation)
+  // let's get rid of sync first
+  // lest stdout's output be intermingled
+  // (otherwise feel free to open a PR for opentelemetry)
+  api.handleSyncBool(input: false, hint: 'bool');
```

Comment:
(not related to the code line, just a general comment)

Could you please try the whole ci yaml, see results, and post a link here so we can predict what will happen when it is merged to the main branch :) 

Btw not sure whether this is helpful: https://github.com/nektos/act

Reply:
I've tried `act` many times, unsuccesfully. it's not 100% the same as running in CI.

Reply:
> Could you please try the whole ci yaml, see results, and post a link here so we can predict what will happen when it is merged to the main branch :)

I'm not sure what you exactly want to achieve : it already run in the ci so why ?

Reply:
Seems no? the non-web is in gh-pages.yaml and not ci.yaml so no results here

Reply:
Ah ok I understand now :) Well we can, but let's just warn contributors because, as explicitly stated in `continuous-benchmark` : _# Do not run this workflow on pull request since this workflow has permission to modify contents._
But indeed if `auto-push` is only `true` for `master`, then we're safe 😃

Reply:
Maybe we can remove "Store benchmark result" step (which modify content) when it is in a PR?